diff --git a/include/spdlog/common.h b/include/spdlog/common.h
index f97fd48c..b1a11e15 100644
--- a/include/spdlog/common.h
+++ b/include/spdlog/common.h
@@ -165,7 +165,13 @@ using remove_cvref_t = typename std::remove_cv<typename std::remove_reference<T>
 template<class T, class Char = char>
 struct is_convertible_to_basic_format_string
     : std::integral_constant<bool,
-          std::is_convertible<T, fmt::basic_string_view<Char>>::value || std::is_same<remove_cvref_t<T>, fmt::basic_runtime<Char>>::value>
+          std::is_convertible<T, fmt::basic_string_view<Char>>::value || std::is_same<remove_cvref_t<T>,
+#if FMT_VERSION >= 100000
+            fmt::runtime_format_string<Char>
+#else
+            fmt::basic_runtime<Char>
+#endif
+          >::value>
 {};
 
 #    if defined(SPDLOG_WCHAR_FILENAMES) || defined(SPDLOG_WCHAR_TO_UTF8_SUPPORT)
